### PR TITLE
kubeadm: add kinder test jobs for k8s 1.20

### PR DESCRIPTION
**Pull request by RA489**
for the kubeadm PR to merge first:
https://github.com/kubernetes/kubeadm/pull/2337

### CODE REVIEW

**Comment by neolit123** [comment]
hm, looks like the release 1.20 dashboards are not created yet:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/19852/pull-test-infra-bazel/1324311666434772992
> annotations: couldn't find dashboard "sig-release-1.20-informing" for job "ci-kubernetes-e2e-kubeadm-kinder-1-20"

we should wait for that first. i asked the question in #release-ci-signal on slack.

**Comment by neolit123** [comment]
@RA489 we are going to have to keep this PR on hold until 1 December when the branch will be created:

https://kubernetes.slack.com/archives/CN0K3TE2C/p1604587020178500

> https://github.com/kubernetes/sig-release/tree/master/releases/release-1.20
> code freeze: Thu November 12
> branch creation: Tue December 1

without the release-1.20 branch these jobs will fail.

**Comment by RA489** [comment]
@neolit123 ready for LGTM.

> @RA489 we are going to have to keep this PR on hold until 1 December when the branch will be created:
> 
> https://kubernetes.slack.com/archives/CN0K3TE2C/p1604587020178500
> 
> > https://github.com/kubernetes/sig-release/tree/master/releases/release-1.20
> > code freeze: Thu November 12
> > branch creation: Tue December 1
> 
> without the release-1.20 branch these jobs will fail.

@neolit123 ready for LGTM

**Comment by justaugustus** [comment]
> without the release-1.20 branch these jobs will fail.

This is _NOT_ ready for approval, as the `release-1.20` does not exist yet.

/hold

**Comment by neolit123** [comment]
ok, the branch was created today:
https://github.com/kubernetes/kubernetes/tree/release-1.20

once we have the `sig-release-1.20-informing` dashboard this PR can merge too:
https://k8s-testgrid.appspot.com/sig-release-master-informing

**Comment by justaugustus** [comment]
Watch this PR: https://github.com/kubernetes/test-infra/pull/20079
I'm working on the branch job creation + dashboard updates there.

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-discovery.yaml [hunk 1]
```diff
@@ -41,6 +41,46 @@ periodics:
...
+    repo: kubernetes
+    base_ref: release-1.20
+    path_alias: k8s.io/kubernetes
+  - org: kubernetes
+    repo: kubeadm
+    base_ref: master
+    path_alias: k8s.io/kubeadm
+  spec:
+    containers:
+    - image: gcr.io/k8s-testimages/kubekins-e2e:v20201031-122dc79-1.20
```
Comment (neolit123) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-discovery.yaml:71]: i don't think this is a valid 1.20 image for kubekins:
https://console.cloud.google.com/gcr/images/k8s-testimages/GLOBAL/kubekins-e2e

@RA489 please change all old and new instances of the string `v20201031-122dc79` to `v20201201-1941b8f`.

Comment (RA489) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-discovery.yaml:71]: sure

Comment (RA489) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-discovery.yaml]: @neolit123 updated the kubekins images.

**Comment by RA489** [comment]
I hope release-1.20 branch is created and we can remove hold on this PR.

**Comment by neolit123** [comment]
> please change all old and new instances of the string v20201031-122dc79 to be v20201201-1941b8f in all the files

we should also update the old images, but i will send a separate PR for that.
thank you for helping with this task!

/lgtm
/approve
/hold cancel

**Pull request merged**